ui/test/AppErrorFallback.test.tsx

```diff
@@ -0,0 +1,47 @@
+import '@testing-library/jest-dom';
+import { render, screen, fireEvent } from '@testing-library/react';
+
+import AppErrorFallback from 'src/components/error/AppErrorFallback.tsx';
+
+describe('AppErrorFallback', () => {
+  const mockError = new Error('Test error message');
+  const mockResetErrorBoundary = jest.fn();
+
+  test('renders the component correctly with all elements and handles language switch', () => {
+    const { container } = render(
+      <AppErrorFallback
+        error={mockError}
+        resetErrorBoundary={mockResetErrorBoundary}
+      />,
+    );
+    const languageButton = screen.getByRole('button', { name: 'Fr' });
+    expect(languageButton).toBeInTheDocument();
+
+    const errorTitle = container.querySelector('.error-title');
+    const errorAdvice = container.querySelector('.error-advice');
+    const homeButton = container.querySelector('a');
+    const errorMessage = container.querySelector('.error-message');
+
+    expect(errorTitle).toHaveTextContent(
+      'The application crashed unexpectedly',
+    );
```

Comment:
It's best to try and think of ways to avoid testing on specific hardcoded text unless it explicitly shouldn't be changed (i.e. if someone edits some text that's in the app for something like a regulatory reason, that should cause a test to fail). Generally, tests that only care about abstract behaviour/functionality, and don't "know" about or rely on things that are just implementation details, are better as they won't start failing/need to be updated for minor reasons (what would be referred to as a "brittle" test). That's desirable because a) a stable test that applies before and after a refactor gives much stronger confidence that behaviour was preserved than a test that is frequently modified at the same time as its target code, and b) being stable means less maintenance work for future devs.

That said, "good testing" (or even good programming) is more art than science, and UI can be tightly coupled to its implementation details. These aren't strict rules that have to always be followed, just generalizations about what is to be preferred :+1:. No need to aim for perfection or stress about it :+1: 

For more applicable advice, assuming my suggested changes to this component are implemented, it should be enough to assert that this component
  1) renders an `h2` with non-empty text content
  2) that the provided error message is rendered somewhere
  3) that there's a link to the home page
  
Tweak that depending on how you decide to handle bilingualism. To know if the biligualism is working, you could pick one specific word like error/erreur that you could reasonably expect to always show up on this page, and use a check on that to assert the correct language is shown. It might still be brittle against a future change to wording, and it's not as strong of an assertion that the specific (current) expected text is rendered, but it should strike a good balance :+1: